shm/Cargo.hpp

```diff
@@ -12,10 +12,11 @@ class Cargo {
     virtual size_t getBasePrice() const = 0;
     virtual size_t getPrice() const = 0;
 
-    Cargo& operator+=(const size_t amount);
-    Cargo& operator-=(const size_t amount);
-    bool operator==(const Cargo& cargo);
-    bool operator!=(const Cargo& cargo);
+    virtual Cargo& operator+=(const size_t amount) = 0;
+    virtual Cargo& operator-=(const size_t amount) = 0;
+
+    virtual bool operator==(const Cargo& cargo) const = 0;
+    bool operator!=(const Cargo& cargo) const;
```

Comment:
If `==` operator is pure virtual then maybe we should have `!=` defined in the same way? What do you think?

Reply:
Yes! I will do this, these are just raw requirements from the task